src/store/plugins/observables.js

```diff
@@ -19,6 +20,31 @@ export default (store) => {
       refCountDelay(1000),
     );
 
+  const balance$ = sdk$.pipe(
+    switchMap((sdk) => timer(0, 3000).pipe(map(() => sdk))),
+    switchMap((sdk) =>
+      sdk
+        ? sdk
+            .balance(store.state.account.publicKey)
+            .catch((error) => {
+              if (!isNotFoundError(error)) {
+                handleUnknownError(error);
+              }
+              return 0;
+            })
+            .then((result) => {
+              const balance = convertToAE(result);
+              if (balance !== store.state.balance) {
+                store.commit('updateBalance', balance);
+              }
+              return balance;
+            })
+        : Promise.resolve(store.state.balance),
+    ),
```

Comment:
We can write it simpler filtering out the `null` values of sdk:
```suggestion
    switchMap((sdk) =>
      sdk.balance(store.state.account.publicKey).catch((error) => {
        if (!isNotFoundError(error)) {
          handleUnknownError(error);
        }
        return 0;
      }),
    ),
    map((balanceAettos) => {
      const balance = convertToAE(balanceAettos);
      if (balance !== store.state.balance) {
        store.commit('updateBalance', balance);
      }
      return balance;
    }),
```
So, proposing to rewrite `sdk$` from the above as:
```js
  const sdk$ = watchAsObservable(({ sdk }) => sdk, { immediate: true }).pipe(
    pluck('newValue'),
    filter((sdk) => sdk),
  );
```